src/ioctls/vcpu.rs

```diff
@@ -1496,6 +1511,147 @@ impl VcpuFd {
+    /// # Example
+    ///
+    ///  ```rust
+    /// # extern crate kvm_ioctls;
+    /// # use kvm_ioctls::{Kvm, SyncReg};
+    /// let kvm = Kvm::new().unwrap();
+    /// let vm = kvm.create_vm().unwrap();
+    /// let mut vcpu = vm.create_vcpu(0).unwrap();
+    /// vcpu.set_sync_dirty_reg(SyncReg::Register);
+    /// ```
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    pub fn set_sync_dirty_reg(&mut self, reg: SyncReg) {
+        let mut kvm_run: &mut kvm_run = self.kvm_run_ptr.as_mut_ref();
+        kvm_run.kvm_dirty_regs |= reg as u64;
+    }
+
+    /// Disable the given [`SyncReg`] to be copied to userspace on the next exit
+    ///
+    /// # Arguments
+    ///
+    /// * `reg` - The [`SyncReg`] to not copy out of the guest
+    ///
+    /// # Example
+    ///
+    ///  ```rust
+    /// # extern crate kvm_ioctls;
+    /// # use kvm_ioctls::{Kvm, SyncReg};
+    /// let kvm = Kvm::new().unwrap();
+    /// let vm = kvm.create_vm().unwrap();
+    /// let mut vcpu = vm.create_vcpu(0).unwrap();
+    /// vcpu.clear_sync_valid_reg(SyncReg::Register);
+    /// ```
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    pub fn clear_sync_valid_reg(&mut self, reg: SyncReg) {
+        let mut kvm_run: &mut kvm_run = self.kvm_run_ptr.as_mut_ref();
+        kvm_run.kvm_valid_regs &= !(reg as u64);
+    }
+
+    /// Tell KVM to not copy the given [`SyncReg`] into the guest on the next entry
+    ///
+    /// # Arguments
+    ///
+    /// * `reg` - The [`SyncReg`] to not copy out into the guest
+    ///
+    /// # Example
+    ///
+    ///  ```rust
+    /// # extern crate kvm_ioctls;
+    /// # use kvm_ioctls::{Kvm, SyncReg};
+    /// let kvm = Kvm::new().unwrap();
+    /// let vm = kvm.create_vm().unwrap();
+    /// let mut vcpu = vm.create_vcpu(0).unwrap();
+    /// vcpu.clear_sync_dirty_reg(SyncReg::Register);
+    /// ```
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    pub fn clear_sync_dirty_reg(&mut self, reg: SyncReg) {
+        let mut kvm_run: &mut kvm_run = self.kvm_run_ptr.as_mut_ref();
+        kvm_run.kvm_dirty_regs &= !(reg as u64);
+    }
+
+    /// Get the [`kvm_sync_regs`] from the VM
+    ///
+    /// # Example
+    ///
+    ///  ```rust
+    /// # extern crate kvm_ioctls;
+    /// # use kvm_ioctls::{Kvm, SyncReg};
+    /// let kvm = Kvm::new().unwrap();
+    /// let vm = kvm.create_vm().unwrap();
+    /// let mut vcpu = vm.create_vcpu(0).unwrap();
+    /// vcpu.set_sync_valid_reg(SyncReg::Register);
+    /// vcpu.run();
+    /// let guest_rax = vcpu.sync_regs().regs.rax;
+    /// ```
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    pub fn sync_regs(&self) -> kvm_sync_regs {
+        let kvm_run: &mut kvm_run = self.kvm_run_ptr.as_mut_ref();
+
+        // SAFETY: Accessing this union field could be out of bounds if the `kvm_run`
+        // allocation isn't large enough. The `kvm_run` region is set using
+        // `get_vcpu_map_size`, to this region is in bounds
```

Comment:
nit:
```suggestion
        // `get_vcpu_map_size`, so this region is in bounds
``` 
Same applies for the method below this one.